### PR TITLE
Clear scanner flags when auto approval is confirmed

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14284

---

I *think* this is enough but this is new to me (and that's why I have to fix this issue 1 week later 😞). I don't know "how" but could we clear the queue? Maybe we could run a migration that sets `needs_human_review_by_mad=False` on all entries?

### CODE REVIEW

File: src/olympia/reviewers/utils.py [hunk 1]
```diff
@@ -943,6 +944,15 @@ def confirm_auto_approved(self):
                 if self.version.needs_human_review:
                     self.version.update(needs_human_review=False)
 
+            # Clear the "needs_human_review" scanner flags too, if any, and
+            # only for the specified version.
+            try:
+                VersionScannerFlags.objects.get(
+                    version=self.version
+                ).clear_needs_human_review_flags()
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/reviewers/utils.py:952]: Ha, right. I was thinking about having a `clear_()` method to only update the logic there if we have a new flag in the future.

[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/reviewers/utils.py]: updated

**Comment by willdurand** [comment]
> Maybe we could run a migration that sets `needs_human_review_by_mad=False` on all entries?

@diox WDYT?

**Comment by diox** [comment]
Yeah, we could even reset by deleting all `VersionScannerFlags` ?

**Comment by willdurand** [comment]
> Yeah, we could even reset by deleting all `VersionScannerFlags` ?

Yep.

File: src/olympia/scanners/migrations/0032_auto_20200519_1438.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Generated by Django 2.2.12 on 2020-05-19 14:38
+
+from django.db import migrations
+
+
+def delete_version_scanner_flags(apps, schema_editor):
+    VersionScannerFlags = apps.get_model('scanners', 'VersionScannerFlags')
+    VersionScannerFlags.objects.all().delete()
```
Comment (diox) [inline src/olympia/scanners/migrations/0032_auto_20200519_1438.py:8]: Add a comment explaining why in case we stumble upon it later...

**Pull request merged**